packages/block-editor/src/components/off-canvas-editor/appender.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { forwardRef } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import Inserter from '../inserter';
+
+export const Appender = forwardRef( ( props, ref ) => {
+	const { hideInserter, clientId } = useSelect( ( select ) => {
+		const {
+			getTemplateLock,
+			__unstableGetEditorMode,
+			getSelectedBlockClientId,
+		} = select( blockEditorStore );
+
+		const _clientId = getSelectedBlockClientId();
```

Comment:
Fetching the clientId internally makes the component more self contained. It also avoids the need for prop drilling on the Offcanvas component itself.

In the future we _may_ want to pass a specific `clientId` at which point we can add an optional prop which will override.

Reply:
TIL. I needed this yesterday!

Reply:
Haha!

Reply:
What you need @scruffian is https://developer.wordpress.org/block-editor/reference-guides/data/data-core-block-editor/